loopd/daemon.go

```diff
@@ -259,6 +229,58 @@ func (d *Daemon) Run() error {
 	return nil
 }
 
+// initializeClient creates and initializes an instance of the swap server
+// client.
+func (d *Daemon) initializeClient() error {
+	// If no swap server is specified, use the default addresses for mainnet
+	// and testnet.
+	if d.cfg.SwapServer == "" {
+		// TODO(wilmer): Use onion service addresses when proxy is
+		// active.
+		switch d.cfg.Network {
+		case "mainnet":
+			d.cfg.SwapServer = mainnetServer
+		case "testnet":
+			d.cfg.SwapServer = testnetServer
+		default:
+			return errors.New("no swap server address specified")
+		}
+	}
+
+	log.Infof("Swap server address: %v", d.cfg.SwapServer)
+
+	// Create an instance of the loop client library.
+	var err error
+	d.impl, d.clientCleanup, err = getClient(d.cfg, &d.lnd.LndServices)
+	if err != nil {
+		return err
+	}
+
+	// Retrieve all currently existing swaps from the database.
+	swapsList, err := d.impl.FetchSwaps()
+	if err != nil {
+		return err
+	}
+
+	for _, s := range swapsList {
+		d.swaps[s.SwapHash] = *s
+	}
+
+	return nil
+}
+
+// stop tries to gracefully shut down the daemon.
+func (d *Daemon) stop() error {
+	if d.lnd != nil {
```

Comment:
Shouldn't this first take down the grpc server?

Reply:
At this point `stop` only contains what was done through the `defer` statements before. The final order will be correct which will be established in later commits.